auth/src/main/java/org/aerogear/mobile/auth/authenticator/oidc/OIDCAuthenticatorImpl.java

```diff
@@ -60,12 +62,14 @@
      * @param authStateManager {@link AuthStateManager}
      * @param authorizationServiceFactory {@link AuthorizationServiceFactory}
      * @param jwksManager {@link JwksManager}
+     * @param certificatePinningCheck {@link CertificatePinningCheck}
      */
     public OIDCAuthenticatorImpl(final ServiceConfiguration serviceConfiguration,
                     final AuthServiceConfiguration authServiceConfiguration,
                     final AuthStateManager authStateManager,
                     final AuthorizationServiceFactory authorizationServiceFactory,
-                    final JwksManager jwksManager) {
+                    final JwksManager jwksManager,
+                    final CertificatePinningCheck certificatePinningCheck) {
```

Comment:
There are getting to be a LOT of parameters on this constructor.  This is seeming like at best a good candidate to adopt a builder or better, or at worst a code smell that it is taking on too many responsibilities.